Mathlib/Topology/Order/Basic.lean

```diff
@@ -2573,6 +2573,95 @@ theorem exists_countable_dense_no_bot_top [SeparableSpace α] [Nontrivial α] :
 
 end DenselyOrdered
 
+section ConditionallyCompleteLinearOrder
+
+variable [ConditionallyCompleteLinearOrder α] [TopologicalSpace α] [OrderTopology α]
+  [ConditionallyCompleteLinearOrder β] [TopologicalSpace β] [OrderClosedTopology β] [Nonempty γ]
+
+/-- Sets are automatically bounded or cobounded in complete lattices. To use the same statements
+in complete and conditionally complete lattices but let automation fill automatically the
+boundedness proofs in complete lattices, we use the tactic `setIsBddDefault` in the statements,
+in the form `(hA : BddAbove A := by setIsBddDefault)`. -/
+
+macro "setIsBddDefault" : tactic =>
+  `(tactic| first
+    | apply OrderTop.bddAbove
+    | apply OrderBot.bddBelow)
+
+/-- A monotone function continuous at the supremum of a nonempty set sends this supremum to
+the supremum of the image of this set. -/
+theorem Monotone.map_sSup_of_continuousAt' {f : α → β} {A : Set α} (Cf : ContinuousAt f (sSup A))
+    (Mf : Monotone f) (A_nonemp : A.Nonempty) (A_bdd : BddAbove A := by setIsBddDefault) :
+    f (sSup A) = sSup (f '' A) := by
+  --This is a particular case of the more general `IsLUB.isLUB_of_tendsto`
+  refine ((@IsLUB.isLUB_of_tendsto α β _ _ _ _ _ _ f A (sSup A) (f (sSup A))
+          (Mf.monotoneOn _) ?_ A_nonemp ?_).csSup_eq (Set.nonempty_image_iff.mpr A_nonemp)).symm
+  · exact isLUB_csSup A_nonemp A_bdd
+  · exact tendsto_nhdsWithin_of_tendsto_nhds Cf
+#align monotone.map_Sup_of_continuous_at' Monotone.map_sSup_of_continuousAt'
+
+/-- A monotone function continuous at the indexed supremum over a nonempty `Sort` sends this indexed
+supremum to the indexed supremum of the composition. -/
+theorem Monotone.map_iSup_of_continuousAt' {ι : Sort _} [Nonempty ι] {f : α → β} {g : ι → α}
+    (Cf : ContinuousAt f (iSup g)) (Mf : Monotone f)
+    (bdd : BddAbove (range g) := by setIsBddDefault) : f (⨆ i, g i) = ⨆ i, f (g i) := by
+  rw [iSup, Monotone.map_sSup_of_continuousAt' Cf Mf (range_nonempty g) bdd, ← range_comp, iSup]
+  rfl
+#align monotone.map_supr_of_continuous_at' Monotone.map_iSup_of_continuousAt'
+
+/-- A monotone function continuous at the infimum of a nonempty set sends this infimum to
+the infimum of the image of this set. -/
+theorem Monotone.map_sInf_of_continuousAt' {f : α → β} {A : Set α} (Cf : ContinuousAt f (sInf A))
+    (Mf : Monotone f) (A_nonemp : A.Nonempty) (A_bdd : BddBelow A := by setIsBddDefault) :
+    f (sInf A) = sInf (f '' A) :=
+  @Monotone.map_sSup_of_continuousAt' αᵒᵈ βᵒᵈ _ _ _ _ _ _ f A Cf Mf.dual A_nonemp A_bdd
```

Comment:
```suggestion
  Monotone.map_sSup_of_continuousAt' (α := αᵒᵈ) (β := βᵒᵈ) Cf Mf.dual A_nonemp A_bdd
```
Let's use our new fancy features!
Same below

Reply:
> Let's use our new fancy features! Same below

There were a lot of `@ _ _ _ _ _` for order dual applications in these files. It turned out easier to search and replace all of them. Unfortunately that increases the diff, this could have been a separate chore PR...

I hope this type of a change is first of all a readability improvement, and second of all more robust. For robustness, instead of the exact number of arguments and order of arguments in the lemmas, with named arguments we now rely on the names of a few arguments (although unfortunately those are often alphas and gammas etc). So code could break if argument names are changed elsewhere in the library, but I think fixing is then easy enough.